include/mata/nfa/nfa.hh

```diff
@@ -276,7 +276,12 @@ public:
      * @return List of transitions leading from @p state_from.
      */
     const Post& get_moves_from(const State state_from) const {
-        assert(state_from < size());
+        const size_t post_size{ size() };
+        if (state_from >= post_size) {
+            throw std::runtime_error(
```

Comment:
Shouldn't we still return the empty post, as before? 
I think every state i delta, source or target, has an allocated post, but states in initial and final maybe don't, and they are actually valid states with empty post, no?

On the other hand, this function will be removed when we have the iterators in delta, right?

Reply:
> Shouldn't we still return the empty post, as before?

We did not return an empty post here. We failed on assertion.

Going forward, we could return an empty post. That would mean that `nfa.delta[1'000'000]` would return an empty post. Is that what we indent to achieve?

> On the other hand, this function will be removed when we have the iterators in delta, right?
No. Why would it be removed because of iterators? You do not want to iterate to some specific state, one state post after another from the beginning. One wants to jump to the state post immediately using a random access function (this one). Its semantics is the same as `operator[]`, which we found confusing (`nfa[4]` really does not explain much, as opposed to `nfa.get_moves_from[4]`) and decided to get rid of in the past. It could be implemented as a random access iterator, using `operator[]`, but then again, that would introduce the `operator[]` function again. This function is supposed to stay as of now.

Reply:
We agreed that the const function should return an empty post, the mutable function should allocate a new `StatePost` for the passed state.

Reply:
Implemented in #292.